pkg/runtime/subscription/postman/http/http.go

```diff
@@ -0,0 +1,402 @@
+			if dlqErr := h.sendBulkToDeadLetter(ctx, bulkSubCallData, &bulkMsg, req.DeadLetterTopic, true); dlqErr == nil {
+				// dlq has been configured and message is successfully sent to dlq.
+				for _, item := range rawMsgEntries {
+					todo.AddBulkResponseEntry(&bsrr.Entries, item.EntryId, nil)
+				}
+				return nil
+			}
+		}
+		bscData.BulkSubDiag.StatusWiseDiag[string(contribpubsub.Retry)] += int64(len(rawMsgEntries))
+
+		for _, item := range rawMsgEntries {
+			todo.AddBulkResponseEntry(&bsrr.Entries, item.EntryId, marshalErr)
+		}
+		return marshalErr
+	}
+
+	spans := make([]trace.Span, len(rawMsgEntries))
+
+	iReq := invokev1.NewInvokeMethodRequest(psm.Path).
+		WithHTTPExtension(nethttp.MethodPost, "").
+		WithRawDataBytes(da).
+		WithContentType(contenttype.JSONContentType).
+		WithCustomHTTPMetadata(psm.Metadata)
+	defer iReq.Close()
+
+	n := 0
+	for _, pubsubMsg := range psm.PubSubMessages {
+		cloudEvent := pubsubMsg.CloudEvent
+		iTraceID := cloudEvent[contribpubsub.TraceParentField]
+		if iTraceID == nil {
+			iTraceID = cloudEvent[contribpubsub.TraceIDField]
+		}
+		if iTraceID != nil {
+			traceID := iTraceID.(string)
+			sc, _ := diag.SpanContextFromW3CString(traceID)
+			var span trace.Span
+			ctx, span = diag.StartInternalCallbackSpan(ctx, "pubsub/"+psm.Topic, sc, h.tracingSpec)
+			if span != nil {
+				spans[n] = span
+				n++
+			}
+		}
+	}
+	spans = spans[:n]
+	defer todo.EndSpans(spans)
+	start := time.Now()
+	resp, err := h.channels.AppChannel().InvokeMethod(ctx, iReq, "")
+	elapsed := diag.ElapsedSince(start)
+	if err != nil {
+		bscData.BulkSubDiag.StatusWiseDiag[string(contribpubsub.Retry)] += int64(len(rawMsgEntries))
+		bscData.BulkSubDiag.Elapsed = elapsed
+		todo.PopulateBulkSubscribeResponsesWithError(psm, &bsrr.Entries, err)
+		return fmt.Errorf("error from app channel while sending pub/sub event to app: %w", err)
+	}
+	defer resp.Close()
+
+	statusCode := int(resp.Status().GetCode())
+
+	for _, span := range spans {
+		m := diag.ConstructSubscriptionSpanAttributes(psm.Topic)
+		diag.AddAttributesToSpan(span, m)
+		diag.UpdateSpanStatusFromHTTPStatus(span, statusCode)
+	}
+
+	if (statusCode >= 200) && (statusCode <= 299) {
+		// Any 2xx is considered a success.
+		var appBulkResponse contribpubsub.AppBulkResponse
+		err = json.NewDecoder(resp.RawData()).Decode(&appBulkResponse)
+		if err != nil {
+			bscData.BulkSubDiag.StatusWiseDiag[string(contribpubsub.Retry)] += int64(len(rawMsgEntries))
+			bscData.BulkSubDiag.Elapsed = elapsed
+			todo.PopulateBulkSubscribeResponsesWithError(psm, &bsrr.Entries, err)
+			return fmt.Errorf("failed unmarshalling app response for bulk subscribe: %w", err)
+		}
+
+		var hasAnyError bool
+		for _, response := range appBulkResponse.AppResponses {
+			if _, ok := (*bscData.EntryIdIndexMap)[response.EntryId]; ok {
+				switch response.Status {
+				case "":
+					// When statusCode 2xx, Consider empty status field OR not receiving status for an item as retry
+					fallthrough
```

Comment:
I noticed the `Deliver` & `DeliverBulk` switch cases handle `""` differently. One falls thru to the retry, but the other falls thru to the success case. Is that intentional?

Reply:
This is the same implementation as before- I agree, but let's change it in a followup.

Reply:
Ill open an issue so we dont lose track here

Reply:
https://github.com/dapr/dapr/issues/8737